ibis/formats/__init__.py

```diff
@@ -0,0 +1,175 @@
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar
+
+from ibis.expr.datatypes import DataType
+from ibis.expr.schema import Schema
+
+T = TypeVar('T')
```

Comment:
I think each of these type variables should have some documentation that gives one or more examples of a concrete type of `T`. For example, (I think) a concrete type for `T` would be `sqlalchemy.sql.type_api.TypeEngine` (the super class of all sqlalchemy types).